search/search_repository.go

```diff
@@ -382,11 +371,66 @@ func (q Query) determineLiteralType(key string, val string) criteria.Expression
 func (q Query) generateExpression() (criteria.Expression, error) {
 	var myexpr []criteria.Expression
 	currentOperator := q.Name
-	if !isOperator(currentOperator) {
-		var key string
-		if val, ok := searchKeyMap[q.Name]; ok {
-			key = val
+
+	// Here we handle the "$WITGROUP" query parameter which we translate from a
+	// simple
+	//
+	// "$WITGROUP = y"
+	//
+	// expression into an
+	//
+	// "Type in (y1, y2, y3, ... ,yn)"
+	//
+	// expression where yi represents the i-th work item type associated with
+	// the work item type group y.
+	handleWitGroup := func(q Query) error {
```

Comment:
If we take this function out, will it help unit test?
If we update the behavior of this function later (not the result of the function) then the test will still pass but we need to maintain this function at two places, I could see it is used in white box tests as well.

Reply:
So you want to only test this function in isolation, right? Well, it is a closure for now so it uses the `myexpr` from above but I guess we can change its signature also. Will try my best to outsource it. Thank you @pranavgore09 . Btw. we have unit tests in the whitebox tests for the `generateExpression` function itself and there I do test this piece (not individually though).